core/imds/src/main/java/software/amazon/awssdk/imds/Ec2Metadata.java

```diff
@@ -29,4 +31,163 @@ public interface Ec2Metadata {
+     * @return The number of seconds for one chunk of data to be read.
+     */
+    Integer httpReadTimeout();
+
+    /**
+     * @return The output stream for debugging.
+     */
+    String httpDebugOutput();
+
+    /**
+     * @return The number of seconds to sleep in-between retries.
+     */
+    Integer backoff();
+
+    /**
+     * @return The SdkHttpClient instance to make the http requests.
+     */
+    SdkHttpClient httpClient();
+
+    /**
+     * @return The Builder Object consisting all the fields.
+     */
+    Ec2Metadata.Builder toBuilder();
+
+    /**
+     * Create an {@code Ec2Metadata} using the default values.
+     */
+    static Ec2Metadata create() {
+        return builder().build();
+    }
+
+    /**
+     * Creates a default builder for {@link Ec2Metadata}.
+     */
+    static Ec2Metadata.Builder builder() {
+        return DefaultEc2Metadata.builder();
+    }
+
+    /**
+     * The builder definition for a {@link Ec2Metadata}.
+     */
+    interface Builder  {
+
+        /**
+         * Define the number of retry attempts for any failed request.
+         *
+         * @param retries The number of retry attempts for any failed request.
+         * @return Returns a reference to this builder
+         */
+        Builder retries(Integer retries);
+
+        /**
+         * Define the endpoint of IMDS.
+         *
+         * @param endpoint The endpoint of IMDS.
+         * @return Returns a reference to this builder
+         */
+        Builder endpoint(String endpoint);
+
+        /**
+         * Define the port for the endpoint.
+         *
+         * @param port The port for the endpoint.
+         * @return Returns a reference to this builder
+         */
+        Builder port(Integer port);
+
+        /**
+         * Define the Time to live (TTL) of the token.
+         *
+         * @param tokenTtl The Time to live (TTL) of the token.
+         * @return Returns a reference to this builder
+         */
+        Builder tokenTtl(Integer tokenTtl);
+
+        /**
+         * Define the endpoint mode of IMDS.Supported values include IPv4 and IPv6.
+         *
+         * @param endpointMode The endpoint mode of IMDS.Supported values include IPv4 and IPv6.
+         * @return Returns a reference to this builder
+         */
+        Builder endpointMode(String endpointMode);
+
+        /**
+         * Define the number of seconds to wait for the connection to open.
+         *
+         * @param httpOpenTimeout TThe number of seconds to wait for the connection to open.
+         * @return Returns a reference to this builder
+         */
+        Builder httpOpenTimeout(Integer httpOpenTimeout);
+
+        /**
+         * Define the number of seconds for one chunk of data to be read.
+         *
+         * @param httpReadTimeout The number of seconds for one chunk of data to be read.
+         * @return Returns a reference to this builder
+         */
+        Builder httpReadTimeout(Integer httpReadTimeout);
```

Comment:
These are configs that are already handled at the `SdkHttpClient` level

Reply:
Noted.